docs/products/opensearch/howto/restore-snapshot-opensearch.md

```diff
@@ -0,0 +1,386 @@
+
+- **S3-compatible services**
+
+  - `endpoint`: Service endpoint for S3-compatible services.
+  - `access_key`: Access key for the S3-compatible service.
+  - `secret_key`**: Secret key for the S3-compatible service.
+  - `region`: Region or endpoint-specific region.
+  - `bucket`, `base_path`, `snapshot_name`: Bucket details and snapshot name.
+  - `server_side_encryption`, `compress`, `chunk_size`: (Optional) Settings
+    for encryption, compression, and file chunking.
+
+## Register the snapshot repository
+
+To begin the restoration process, register the snapshot repository with your
+Aiven for OpenSearch service. This step involves configuring your service to recognize
+and access the storage location where your OpenSearch or Elasticsearch snapshots are
+stored.
+
+### Google Cloud Storage (GCS)
+
+Register a snapshot repository on GCS.
+
+<Tabs groupId="method">
+<TabItem value="api" label="API" default>
+
+```bash
+curl -X PUT "https://SERVICE_NAME.aivencloud.com/_snapshot/my-snapshot-repo" \
+-H "Authorization: Bearer API_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+  "type": "gcs",
+  "settings": {
+    "bucket": "my-gcs-bucket",
+    "base_path": "snapshots/",
+    "credentials": "GCS_CREDENTIALS_FILE_CONTENT"
+  }
+}'
+```
+
+</TabItem>
+<TabItem value="cli" label="CLI">
+
+```bash
+avn service update \
+  --project PROJECT_NAME \
+  --service SERVICE_NAME \
+  -c gcs_migration.bucket="my-gcs-bucket" \
+  -c gcs_migration.base_path="snapshots/" \
+  -c gcs_migration.credentials="GCS_CREDENTIALS_FILE_CONTENT" \
+  -c gcs_migration.snapshot_name="SNAPSHOT_NAME"
+
+```
+
+</TabItem>
+</Tabs>
+
+### Amazon S3
+
+Register a snapshot repository on Amazon S3.
+
+<Tabs groupId="method">
+<TabItem value="api" label="API" default>
+
+```bash
+curl -X PUT "https://SERVICE_NAME.aivencloud.com/_snapshot/my-snapshot-repo" \
+-H "Authorization: Bearer API_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+  "type": "s3",
+  "settings": {
+    "bucket": "my-bucket",
+    "region": "us-west-2"
+  }
+}'
+```
+
+</TabItem>
+<TabItem value="cli" label="CLI">
+
+```bash
+avn service update \
+  --project PROJECT_NAME \
+  --service SERVICE_NAME \
+  -c s3_migration.bucket="my-bucket" \
+  -c s3_migration.region="us-west-2" \
+  -c s3_migration.base_path="snapshots/" \
+  -c s3_migration.access_key="your-access-key" \
+  -c s3_migration.secret_key="your-secret-key" \
+  -c s3_migration.snapshot_name="SNAPSHOT_NAME"
+
+```
+
+</TabItem>
+</Tabs>
+
+### Microsoft Azure
+
+Register a snapshot repository on Microsoft Azure.
+
+<Tabs groupId="method">
+<TabItem value="api" label="API" default>
+
+```bash
+curl -X PUT "https://SERVICE_NAME.aivencloud.com/_snapshot/my-snapshot-repo" \
+-H "Authorization: Bearer API_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+  "type": "azure",
+  "settings": {
+    "container": "my-container",
+    "base_path": "snapshots/",
+    "account": "my-account",
+    "key": "your-key"
+  }
+}'
+```
```

Comment:
I don't think this steps (for each cloud provider) are needed and even possible from customer's machine. 
Everything including creation of repository is done by our automation, so customer should just update user-config parameters with credentials according to our current logic.